Jellyfin.Server.Implementations/Devices/DeviceManager.cs

```diff
@@ -110,21 +110,21 @@ public ClientCapabilities GetCapabilities(string deviceId)
         /// <inheritdoc />
         public async Task<DeviceInfo?> GetDevice(string id)
         {
-            Device? device;
             var dbContext = await _dbProvider.CreateDbContextAsync().ConfigureAwait(false);
             await using (dbContext.ConfigureAwait(false))
             {
-                device = await dbContext.Devices
+                var device = await dbContext.Devices
                     .Where(d => d.DeviceId == id)
                     .OrderByDescending(d => d.DateLastActivity)
                     .Include(d => d.User)
+                    .SelectMany(d => dbContext.DeviceOptions.Where(o => o.DeviceId == d.DeviceId).DefaultIfEmpty(), (d, o) => new { Device = d, Options = o })
```

Comment:
Does EF actually know how to build the SQL expression without a navigation property? I feel like this causes many lookups instead of a join

Reply:
Using the ToQueryString function, EF core generates

SELECT "d"."Id", "d"."AccessToken", "d"."AppName", "d"."AppVersion", "d"."DateCreated", "d"."DateLastActivity", "d"."DateModified", "d"."DeviceId", "d"."DeviceName", "d"."IsActive", "d"."UserId", "u"."Id", "u"."AudioLanguagePreference", "u"."AuthenticationProviderId", "u"."DisplayCollectionsView", "u"."DisplayMissingEpisodes", "u"."EnableAutoLogin", "u"."EnableLocalPassword", "u"."EnableNextEpisodeAutoPlay", "u"."EnableUserPreferenceAccess", "u"."HidePlayedInLatest", "u"."InternalId", "u"."InvalidLoginAttemptCount", "u"."LastActivityDate", "u"."LastLoginDate", "u"."LoginAttemptsBeforeLockout", "u"."MaxActiveSessions", "u"."MaxParentalAgeRating", "u"."MustUpdatePassword", "u"."Password", "u"."PasswordResetProviderId", "u"."PlayDefaultAudioTrack", "u"."RememberAudioSelections", "u"."RememberSubtitleSelections", "u"."RemoteClientBitrateLimit", "u"."RowVersion", "u"."SubtitleLanguagePreference", "u"."SubtitleMode", "u"."SyncPlayAccess", "u"."Username", "d0"."Id", "d0"."CustomName", "d0"."DeviceId"

FROM "Devices" AS "d"

LEFT JOIN "DeviceOptions" AS "d0" ON "d"."DeviceId" = "d0"."DeviceId"

INNER JOIN "Users" AS "u" ON "d"."UserId" = "u"."Id"

ORDER BY "d"."DateLastActivity" DESC, "d"."DeviceId"

Reply:
Neat